app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/meta/AggregateMetadataHandler.java

```diff
@@ -48,6 +49,26 @@ public boolean canHandle(StepKind kind) {
         return StepKind.aggregate.equals(kind);
     }
 
+    @Override
+    public DynamicActionMetadata createMetadata(Step step, List<Step> previousSteps, List<Step> subsequentSteps) {
+        DataShape inputShape = subsequentSteps.stream()
+                                            .filter(StepMetadataHelper::hasInputDataShape)
+                                            .findFirst()
+                                            .map(StepMetadataHelper::getInputDataShape)
+                                            .orElse(StepMetadataHelper.NO_SHAPE);
```

Comment:
Do you think it would make sense to add these methods to `DataShape`, something like `static Optional<DataShape> findFirstInputIn(List<WithDataShape> items)` (introducting `WithDataShape`). I kinda like when APIs are readable, makes the intent clearer.